### ISSUE TITLE
Core dump when installing from test repos

### ISSUE THREAD

**Issue by mbooth101** [flatpak/flatpak#2727]
## Distro and Versions

Fedora 29

```
$ rpm -qa flatpak
flatpak-1.2.3-2.fc29.x86_64
```

## Description of the problem

Core dump when trying to install a test build. I was following the instruction given by the bot in this comment: https://github.com/flathub/flathub/pull/870#issuecomment-466945551

And this is the error I recieved:

```
$ flatpak install --user https://dl.flathub.org/build-repo/300/org.freedesktop.Sdk.Extension.openjdk.flatpakref
**
flatpak:ERROR:app/flatpak-main.c:700:flatpak_run: assertion failed: (success || error)
Aborted (core dumped)
```

## Steps to reproduce

Just follow the testing instructions given by the build bot:

```
$ flatpak install --user https://dl.flathub.org/build-repo/300/org.freedesktop.Sdk.Extension.openjdk.flatpakref
```

**Comment by mbooth101** [comment]
Might be important: This is a SDK Extension, not an actual application. While it might not be supported to install an SDK Extension this way, it definitely shouldn't core dump :-)

**Comment by mbooth101** [comment]
Backtrace:

```
(gdb) bt
#0  0x00007ffff74d653f in raise () at /lib64/libc.so.6
#1  0x00007ffff74c0895 in abort () at /lib64/libc.so.6
#2  0x00007ffff76bfb53 in  () at /lib64/libglib-2.0.so.0
#3  0x00007ffff771a4de in g_assertion_message_expr () at /lib64/libglib-2.0.so.0
#4  0x000055555559ed57 in flatpak_run (res_error=0x7fffffffcf00, argv=<optimized out>, argc=<optimized out>) at app/flatpak-main.c:700
#5  0x000055555559ed57 in main (argc=<optimized out>, argv=<optimized out>) at app/flatpak-main.c:801
```

**Comment by mwleeds** [comment]
The issue is that URL is a 404, which Flatpak apparently can't handle gracefully

**Comment by AsciiWolf** [comment]
Could this fix be also backported to the flatpak-1.2.x branch?
